### ISSUE TITLE
RuntimeError!

### ISSUE THREAD

**Issue by SomnusQue** [microsoft/Cream#219]
I run auto_100weight_inherit_100to75.sh, and meet this problem. I think I have been ready everything for this project, but it still have some problems which I can't solve. Please somebody help me!

**Comment by SomnusQue** [comment]
![Uploading 2859391705835847_.pic_hd.jpg…]()

**Comment by SomnusQue** [comment]
![Uploading 2859391705835847.jpg…]()

**Comment by SomnusQue** [comment]
![2859391705835847](https://github.com/microsoft/Cream/assets/57973825/b38a5113-3304-43b7-891b-8cef045d41cf)

**Comment by wkcn** [comment]
Hi @SomnusQue , thanks for your attention to our work!

Is the code of TinyCLIP latest?

It is a bug which is triggered on PyTorch 2.x.
We have fixed the bug by adding this line: https://github.com/microsoft/Cream/blob/main/TinyCLIP/src/open_clip/model.py#L28

```python
checkpoint = functools.partial(checkpoint, use_reentrant=False)
```

**Comment by SomnusQue** [comment]
> Hi @SomnusQue , thanks for your attention to our work!
> 
> Is the code of TinyCLIP latest?
> 
> It is a bug which is triggered on PyTorch 2.x. We have fixed the bug by adding this line: https://github.com/microsoft/Cream/blob/main/TinyCLIP/src/open_clip/model.py#L28
> 
> ```python
> checkpoint = functools.partial(checkpoint, use_reentrant=False)
> ```

OMG! The author answer my question! The code which I have really doesn't have these lines! Thx for your patience!
But I wondering when is the code update?

**Comment by SomnusQue** [comment]
> > Hi @SomnusQue , thanks for your attention to our work!
> > Is the code of TinyCLIP latest?
> > It is a bug which is triggered on PyTorch 2.x. We have fixed the bug by adding this line: https://github.com/microsoft/Cream/blob/main/TinyCLIP/src/open_clip/model.py#L28
> > ```python
> > checkpoint = functools.partial(checkpoint, use_reentrant=False)
> > ```
> 
> OMG! The author answer my question! The code which I have really doesn't have these lines! Thx for your patience! But I wondering when is the code update?
Furthermore... Is this LOSS normal?
![2859731705841213_ pic_hd](https://github.com/microsoft/Cream/assets/57973825/4d9c57bc-0fa8-40fb-bdda-615a3e0c7562)

**Comment by wkcn** [comment]
@SomnusQue I fixed the bug in Jan. 11, 2024 (https://github.com/microsoft/Cream/pull/218/files#diff-2c756c8b8b99609dee1b59ce4dcfaf773aa9afbc84e093e03e3e0de653fa0124R28).

You can visualize the loss curve in wandb. The loss is normal if it is decreasing : )

**Comment by SomnusQue** [comment]
> @SomnusQue I fixed the bug in Jan. 11, 2024 (https://github.com/microsoft/Cream/pull/218/files#diff-2c756c8b8b99609dee1b59ce4dcfaf773aa9afbc84e093e03e3e0de653fa0124R28).
> 
> You can visualize the loss curve in wandb. The loss is normal if it is decreasing : )

Thanks for your patience! Due to the cluster, I can't use wandb(because it needs network..?), I change this line in .sh file'--report-to wandb' to '--report-to tensorboard'. Does it have anywhere else need to change in the code?

**Comment by wkcn** [comment]
@SomnusQue 
No code change required. It is also available to set the environmental variable `WANDB_MODE=offline`. The wandb log will be saved as a file. Then run `wandb sync <file path>` to upload the log.

**Comment by SomnusQue** [comment]
> @SomnusQue
> No code change required. It is also available to set the environmental variable `WANDB_MODE=offline`. The wandb log will be saved as a file. Then run `wandb sync <file path>` to upload the log.

sry to bother u again...
![3971705891663_ pic_hd](https://github.com/microsoft/Cream/assets/57973825/7281f5cc-2984-4564-86ff-70a2e55b3289)
The result in tensorboard seems like sth went wrong...
![3981705892612_ pic_hd](https://github.com/microsoft/Cream/assets/57973825/de4a211e-9dfb-47ce-acd0-c24247090cde)
This is the final epoch of my training result..

**Comment by SomnusQue** [comment]
![3991705910386_ pic](https://github.com/microsoft/Cream/assets/57973825/9835b3c6-de43-4aa2-a9c0-6792a3700cb3)
![4001705910406_ pic](https://github.com/microsoft/Cream/assets/57973825/e50392ec-7753-4c19-b0cb-f93071c12333)
This is our bash file, is there sth wrong...?

**Comment by wkcn** [comment]
Sorry that I did not test TensorBoard yet.

The training data in the provided script is synthetic.
They should be replaced with the following command:
```bash
 --train-data <your yfcc_path or laion_path/> \
 --dataset-type webdataset \
```

**Comment by SomnusQue** [comment]
> Sorry that I did not test TensorBoard yet.
> 
> The training data in the provided script is synthetic. They should be replaced with the following command:
> 
> ```shell
>  --train-data <your yfcc_path or laion_path/> \
>  --dataset-type webdataset \
> ```

The training data should the path where I store ImageNet?

**Comment by SomnusQue** [comment]
> Sorry that I did not test TensorBoard yet.
> 
> The training data in the provided script is synthetic. They should be replaced with the following command:
> 
> ```shell
>  --train-data <your yfcc_path or laion_path/> \
>  --dataset-type webdataset \
> ```

I downloaded laion file, and put it in the path '/.cache/clip/'.
Is this the path I need to write?

**Comment by wkcn** [comment]
@SomnusQue 
Please refer to the document https://github.com/mlfoundations/open_clip?tab=readme-ov-file#data
